civis/base.py

```diff
@@ -84,3 +85,81 @@ def _call_api(self, method, path=None, params=None, data=None, **kwargs):
             resp = convert_response_data_type(resp,
                                               return_type=self._return_type)
             return resp
+
+
+FINISHED = ['success', 'succeeded']
+FAILED = ['failed']
+NOT_FINISHED = ['queued', 'running']
+CANCELLED = ['cancelled']
+DONE = FINISHED + FAILED + CANCELLED
+
+# Translate Civis state strings into `future` state strings
+STATE_TRANS = {}
+for name in FINISHED + FAILED:
+    STATE_TRANS[name] = futures._base.FINISHED
+for name in NOT_FINISHED:
+    STATE_TRANS[name] = futures._base.RUNNING
+for name in CANCELLED:
+    STATE_TRANS[name] = futures._base.CANCELLED_AND_NOTIFIED
+
+
+class CivisAsyncResultBase(futures.Future):
```

Comment:
Yeah, that makes sense. `_check_result` doesn't have to be overwritten by subclasses, by default it will just check the `_result` attribute. It is overwritten by `PollableResult` because it checks by polling where the check isn't overwritten on `SubscribableResult` so it just checks if something has been set on `._result`. 

`set_result` or `set_exception` should be called and if there are any intermediate states that need to be specified than either setting `_result` or overwriting `_check_result` can be used to accomplish that. I'll draft up a docstring for the Base class describing how it should be used by subclasses.